content/en/blog/github-swiss-knife-3/index.md

```diff
@@ -0,0 +1,203 @@
+and an excellent source for discussion. It is easy to refer to articles that even date back some
+years when all of the posts are collected on the same site.
+
+We use [the Hugo framework](https://gohugo.io/) for building the site, specifically
+[the Docsy theme](https://www.docsy.dev/),
+which is especially suitable for software project documentation. The Hugo framework uses
+a low-code approach, where most of the content can be edited as Markdown files.
+However, when needed, modifications can also be made to the HTML pages quite easily.
+
+{{< imgproc cover Fit "825x825" >}}
+<em>The project site serves as a platform for documentation and blogs.
+</em>
+{{< /imgproc >}}
+
+We publish the site via [GitHub Pages](https://pages.github.com/),
+a static website hosting solution that
+GitHub offers for each repository. When the Pages feature is enabled, GitHub serves
+the content of the pages from a dedicated branch.
+[A GitHub action handles the deployment](https://github.com/findy-network/findy-network.github.io/blob/f72bcd9cdec4882eae541320f5dce48499965d4f/.github/workflows/on-release.yml#L7),
+i.e., builds the website and pushes the site files to this dedicated branch. The site URL
+combines organization and repository names. An exception is the `<organisation-name> .github.io`-named
+repositories that one can find in the similarly formatted URL, e.g., our
+[`findy-network/findy-network.github.io`](https://github.com/findy-network/findy-network.github.io)
+repository is served in URL <https://findy-network.github.io>.
+
+## Dependabot
+
+[Dependabot](https://docs.github.com/en/code-security/dependabot) is a GitHub-offered service
+that helps developers with the security and
+version updates of repository dependencies. When Dependabot finds a vulnerable dependency
+that can be replaced with a more secure version, it creates a pull request that implements
+this change. A similar approach exists with dependency version updates:
+When a new version is released, Dependabot creates a PR with the dependency update to the new version.
+
+{{< imgproc dependabot Fit "825x825" >}}
+<em>Dependabot analyzes the project dependencies based on the dependency graph.
+It creates PRs for security and version updates automatically.
+</em>
+{{< /imgproc >}}
+
+The pull requests that the Dependabot creates are like any other PR.
+The CI runs tests on Dependabot PRs similar to those, as with the changes developers have made.
+One can review and merge the PR similarly.
+
+Using Dependabot requires a comprehensive enough automated test set. The one who merges
+the PR needs to confirm that the change will not cause regression. Manual testing is required
+if the CI is missing the automated tests for the changes. If one needs manual testing, the benefit
+of using the bot for automated PR creation is lost, and it might even be more laborious to update
+the dependencies one by one.
+
+In our project, we used the principle of testing all the crucial functionality with
```

Comment:
The chapter is well-written and vital. Should it be highlighted somehow? Unfortunately, nothing came to my mind. Maybe learning snippet to the end: 

> You should estimate the risk/effort ratio when automating crucial tasks.